tests/testthat/helpers-with-settings.R

```diff
@@ -31,6 +31,57 @@ app_tws <- function() {
   )
 }
 
+#' Plot with settings app
+#'
+#' @description Example plot with setting app for testing using \code{shinytest2}
+#'
+#' @keywords internal
+#'
+app_pws <- function() {
+  shiny::shinyApp(
+    ui = shiny::fluidPage(
+      shinyjs::useShinyjs(),
+      shiny::actionButton("button", "Show/Hide"),
+      plot_with_settings_ui(
+        id = "plot_with_settings"
+      )
+    ),
+    server = function(input, output, session) {
+      plot_r <- shiny::reactive({
+        ggplot2::ggplot(data.frame(x = 1:5, y = 1:5)) +
+          ggplot2::geom_point(ggplot2::aes(x = 1:5, y = 1:5))
+      })
+
+      show_hide_signal <- shiny::reactiveVal(TRUE)
+
+      shiny::observeEvent(input$button, {
+        show_hide_signal(
+          shiny::isolate(
+            !show_hide_signal()
+          )
```

Comment:
`isolate` not needed as `observeEvent` doesn't observe the `show_hide_signal` but the `input$button`
```suggestion
          !show_hide_signal()
```